modules/appeals_api/app/models/concerns/appeals_api/hlr_status.rb

```diff
@@ -22,7 +23,9 @@ module HlrStatus
 
       def versioned_statuses
         case api_version.downcase
-        when 'V2'
+        when 'v0'
+          V0_STATUSES
+        when 'v2'
```

Comment:
Can I get an extra set of eyes on this? Before this change, `api_version.downcase` would never have matched "V2," so the V1 statuses would have been returned every time. If you look at the bottom of this file, the `status` field is validated based on the results of this method, so I want to be sure that we have it right.

Reply:
What you've said here sounds right to me, too - it seems like it would be nice to settle on consistently using upcase or downcase for all `v*` strings we work with and convert them all at some point if possible (but not as part of this ticket/PR)

Reply:
Agreed! I went back and forth on upcase vs. downcase a few times since we use both.

I investigated the supported statuses a bit more and am comfortable with this change now. Before this change, we were technically allowing the "uploaded", "received", and "expired" statuses for V2 and V0 HLRs, even though those statuses are not supported in V2 or V0 according to our documentation. To be extra sure, I also confirmed in Production that we don't have any V2 or V0 HLR records that are in those statuses.

Before:
V1 HLR: ["pending", "submitting", "submitted", "processing", "error", "uploaded", "received", "success", "expired", "complete"]
V2 HLR: ["pending", "submitting", "submitted", "processing", "error", "uploaded", "received", "success", "expired", "complete"]
V0 HLR: ["pending", "submitting", "submitted", "processing", "error", "uploaded", "received", "success", "expired", "complete"]

After:
V1 HLR: ["pending", "submitting", "submitted", "processing", "error", "uploaded", "received", "success", "expired", "complete"]
V2 HLR: ["pending", "submitting", "submitted", "processing", "success", "complete", "error"]
V0 HLR: ["pending", "submitting", "submitted", "processing", "success", "complete", "error"]

Also, I just added some specs to test that the correct supported HLR statuses are returned for each API version.